code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -323,14 +323,15 @@
 	else if(istype(KA.loc, /mob/living/basic/mining_drone))
 		to_chat(user, span_notice("The modkit you're trying to install is not rated for minebot use."))
 		return FALSE
-	if(denied_type)
-		var/number_of_denied = 0
-		for(var/obj/item/borg/upgrade/modkit/modkit_upgrade as anything in KA.modkits)
-			if(istype(modkit_upgrade, denied_type))
-				number_of_denied++
-			if(number_of_denied >= maximum_of_type)
-				. = FALSE
-				break
+	if(!denied_type)
```

Comment:
I feel like this is on me for not being more specific. There were two cases for this fallback. If ``denied_type`` is null and ``maximum_of_type`` is above 0.

The way you have implemented this has created the same foot gun but in reverse. Especially since you have had to add 99 to the baseline mods, but this will change game behaviour as there are some lavaland loot mods that are expected to be above a maximum of 1. Like lifesteal.

To fix this:
A) keep this if check with this suggestion above the original code that was here.
```suggestion
	if(!denied_type && maxium_of_type)
```
B) Revert the changes to having to needing to pass the if check for denied type before looking to see if we need to limit the mods.

C) Define the default foe maximum_of_type to 0.